src/main/java/seedu/duke/commands/Add.java

```diff
@@ -0,0 +1,125 @@
+        if (endIndex == -1) {
+            return input.substring(startIndex);
+        } else {
+            return input.substring(startIndex, endIndex);
+        }
+    }
+
+    /**
+     * Function to check if the format of input is correct or not
+     * @param input input entered by user. Format: String
+     * @throws InvalidInputFormatException exception thrown if format of input is incorrect
+     */
+
+    public void checkFormat(String input) throws InvalidInputFormatException {
+        boolean isRight;
+        isRight = InvalidInputFormatException.checkMod(input);
+        checkFormatException(isRight);
+        isRight = InvalidInputFormatException.checkSem(input);
+        checkFormatException(isRight);
+        isRight = InvalidInputFormatException.checkMC(input);
+        checkFormatException(isRight);
+        isRight = InvalidInputFormatException.checkGrade(input);
+        checkFormatException(isRight);
+    }
+
+    public void checkFormatException(boolean isRight) throws InvalidInputFormatException {
+        if (!isRight) {
+            throw new InvalidInputFormatException();
+        }
+    }
+
+    /**
+     * Function to check if content entered by user is empty or not
+     * @param input input entered by user. Format: String
+     * @param idx a collection of indexes where the details should be present. If these are empty, an exception should be thrown
+     * @throws InvalidInputContentException exception thrown if content of input is empty
+     */
+    public void checkContent(String input, int[] idx) throws InvalidInputContentException {
+        boolean isSame;
+        isSame = InvalidInputContentException.emptyContent(idx[0], idx[1], input);
+        checkContentException(isSame);
+        isSame = InvalidInputContentException.emptyContent(idx[2], idx[3], input);
+        checkContentException(isSame);
+        isSame = InvalidInputContentException.emptyContent(idx[4], idx[5], input);
+        checkContentException(isSame);
+        isSame = InvalidInputContentException.emptyContent(idx[6], idx[7], input);
+        checkContentException(isSame);
+    }
+
+    public void checkContentException(boolean isSame) throws InvalidInputContentException {
+        if (isSame) {
+            throw new InvalidInputContentException();
+        }
+    }
+
+    /**
+     * function to return the positions of the details in input
+     * @param input the input given by user. Format: String
+     * @return an integer array containing the positions of the details given by user
+     */
+    public int[] positions(String input) {
+        int[] idx = new int[8];
+        idx[0] = input.indexOf("m/") + 2;
+        idx[1] = input.indexOf(" ", idx[0]);
+        idx[2] = input.indexOf("s/") + 2;
+        idx[3] = input.indexOf(" ", idx[2]);
+        idx[4] = input.indexOf("mc/") + 3;
+        idx[5] = input.indexOf(" ", idx[4]);
+        idx[6] = input.indexOf("g/") + 2;
+        idx[7] = input.indexOf(" ", idx[6]);
```

Comment:
Again, lots of  magic liertals. Consider moving them into different constant variables to make the code more understandable.